stdlib/3/re.pyi

```diff
@@ -5,105 +5,147 @@
 # based on: http://docs.python.org/3.2/library/re.html
 # and http://hg.python.org/cpython/file/618ea5612e83/Lib/re.py
 
+import sys
 from typing import (
     List, Iterator, overload, Callable, Tuple, Sequence, Dict,
     Generic, AnyStr, Match, Pattern, Any, Optional, Union
 )
 
 # ----- re variables and constants -----
-A = 0
-ASCII = 0
-DEBUG = 0
-I = 0
-IGNORECASE = 0
-L = 0
-LOCALE = 0
-M = 0
-MULTILINE = 0
-S = 0
-DOTALL = 0
-X = 0
-VERBOSE = 0
-U = 0
-UNICODE = 0
-T = 0
-TEMPLATE = 0
+if sys.version_info >= (3, 6):
+    import enum
+    class RegexFlag(enum.IntFlag):
+        A = 0
+        ASCII = 0
+        DEBUG = 0
+        I = 0
+        IGNORECASE = 0
+        L = 0
+        LOCALE = 0
+        M = 0
+        MULTILINE = 0
+        S = 0
+        DOTALL = 0
+        X = 0
+        VERBOSE = 0
+        U = 0
+        UNICODE = 0
+        T = 0
+        TEMPLATE = 0
+
+    A = RegexFlag.A
+    ASCII = RegexFlag.ASCII
+    DEBUG = RegexFlag.DEBUG
+    I = RegexFlag.I
+    IGNORECASE = RegexFlag.IGNORECASE
+    L = RegexFlag.L
+    LOCALE = RegexFlag.LOCALE
+    M = RegexFlag.M
+    MULTILINE = RegexFlag.MULTILINE
+    S = RegexFlag.S
+    DOTALL = RegexFlag.DOTALL
+    X = RegexFlag.X
+    VERBOSE = RegexFlag.VERBOSE
+    U = RegexFlag.U
+    UNICODE = RegexFlag.UNICODE
+    T = RegexFlag.T
+    TEMPLATE = RegexFlag.TEMPLATE
+    _FlagsType = RegexFlag
```

Comment:
I think this should be `Union[int, _FlagsType]`, since people can still pass ints directly. (In particular, I suspect there's code out there that just passes `0` as the `flags` argument to some re functions.)

Reply:
Good point.